### ISSUE TITLE
Deployment to Windows app service has stopped working in 1.9.4

### ISSUE THREAD

**Issue by TWolversonReply** [Azure/azure-dev#4071]
- [ ] Make sure you've installed the latest version using [instructions in the wiki](../wiki/install)

**Output from `azd version`**
1.9.4

**Describe the bug**
Between 1.9.3 and 1.9.4, deployment of a .NET web app to a Windows app service has started failing with the error message

`ERROR: failed deploying service 'web': deploying service web: empty deployment status id`

The code being deployed is from a universal package on the same version as before, so I'm certain it isn't our code change reponsible, since there hasn't been any.

**Expected behavior**
Deployment works

**Environment**
Information on your environment:
    * Language name and version
    * IDE and version : [e.g. Visual Studio 16.3]

**Additional context**

I would imagine this has been introduced by https://github.com/Azure/azure-dev/pull/3965 Add deployment status tracking for linux web apps.

Fixing the installed azd version in the ADO install-azd pipeline component to 1.9.3 fixes the problem.

**Comment by rajeshkamal5050** [comment]
@hemarina can you look into this? cc @weikanglim @vhvb1989

**Comment by jongio** [comment]
I switched todo-csharp-sql to windows and I couldn't repro this - it deployed.  @TWolversonReply it would be helpful to get more context on how you've structured your app so we can try to get a repro.

**Comment by WorkJelle** [comment]
I had this problem aswell with Logic app Standard. My serverplan is on a different resourcegroup than where my logic app standards are hosted. So during my provisioning state everything seems to be going well. In the json view you can see that the serverfarmId is correctly configured.

````json
"/subscriptions/<subId>/resourceGroups/shared-dev/providers/Microsoft.Web/serverfarms/plan-logic-shared-dev"
````

However during the deploy state, where the files get packaged and sent it will give out the error:

````json
{
  "error": {
    "code": "ResourceNotFound",
    "message": "The Resource 'Microsoft.Web/serverFarms/plan-logic-shared-dev' under resource group 'd365fscm-dev' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix"
  }
}
````

d365fscm-dev resourcegroup = the place where my logic app standard will reside
Shared-dev resourcegroup = the place where the app service resides 

During the deploy step in the deployment you can see that it tries to do a GET request towards, where the serverfarm isn't even located.:

````json
"https://management.azure.com/subscriptions/<subId>/resourceGroups/d365fscm-dev/providers/Microsoft.Web/serverfarms/plan-logic-shared-dev"
````

My temporary fix now is just to use version 1.9.3 to atleast make my deployments work
